File: src/types/system.ts
export interface GpuInfo {
  name: string
  vram_gb: number
  compute_capability?: string
  cuda_version?: string
  driver_version?: string
}

export interface SystemInfo {
  gpu: GpuInfo | null
  has_gpu: boolean
  cpu: string
  ram_gb: number
  os: string
  python_version: string
  disk_free_gb: number
}

export interface ModelCompatibility {
  name: string
  parameters: string
  can_train: boolean
  can_infer: boolean
  vram_training_gb: number
  vram_inference_gb: number
  license: string
  description: string
  gated: boolean
  reason?: string
}

File: python/api/schemas/system.py
from pydantic import BaseModel
from typing import Optional


class GpuInfo(BaseModel):
    name: str
    vram_gb: float
    compute_capability: Optional[str] = None
    cuda_version: Optional[str] = None
    driver_version: Optional[str] = None


class SystemInfo(BaseModel):
    gpu: Optional[GpuInfo] = None
    has_gpu: bool
    cpu: str
    ram_gb: float
    os: str
    python_version: str
    disk_free_gb: float


class ModelCompatibility(BaseModel):
    name: str
    parameters: str
    can_train: bool
    can_infer: bool
    vram_training_gb: int
    vram_inference_gb: int
    license: str
    description: str
    gated: bool = False
    reason: Optional[str] = None
